sdk/storage/inc/adls/protocol/adls_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+  {
+    case PathLeaseAction::Acquire:
+      return "acquire";
+    case PathLeaseAction::Break:
+      return "break";
+    case PathLeaseAction::Change:
+      return "change";
+    case PathLeaseAction::Renew:
+      return "renew";
+    case PathLeaseAction::Release:
+      return "release";
+    default:
+      return std::string();
+  }
+}
+
+static PathLeaseAction PathLeaseActionFromString(const std::string& pathLeaseAction)
+{
+  if (pathLeaseAction == "acquire")
+  {
+    return PathLeaseAction::Acquire;
+  }
+  if (pathLeaseAction == "break")
+  {
+    return PathLeaseAction::Break;
+  }
+  if (pathLeaseAction == "change")
+  {
+    return PathLeaseAction::Change;
+  }
+  if (pathLeaseAction == "renew")
+  {
+    return PathLeaseAction::Renew;
+  }
+  if (pathLeaseAction == "release")
+  {
+    return PathLeaseAction::Release;
+  }
+  throw std::runtime_error("Cannot convert " + pathLeaseAction + " to PathLeaseAction");
+};
+
+// Optional. If the value is "getStatus" only the system defined properties for the path are returned. If the value is "getAccessControl" the access control list is returned in the response headers (Hierarchical Namespace must be enabled for the account), otherwise the properties are returned.
+enum class PathGetPropertiesAction
+{
+  GetAccessControl,
+  GetStatus,
+  Unknown
+};
+
+static std::string PathGetPropertiesActionToString(const PathGetPropertiesAction& pathGetPropertiesAction)
+{
+  switch (pathGetPropertiesAction)
+  {
+    case PathGetPropertiesAction::GetAccessControl:
+      return "getAccessControl";
+    case PathGetPropertiesAction::GetStatus:
+      return "getStatus";
+    default:
+      return std::string();
+  }
+}
+
+static PathGetPropertiesAction PathGetPropertiesActionFromString(const std::string& pathGetPropertiesAction)
+{
+  if (pathGetPropertiesAction == "getAccessControl")
+  {
+    return PathGetPropertiesAction::GetAccessControl;
+  }
+  if (pathGetPropertiesAction == "getStatus")
+  {
+    return PathGetPropertiesAction::GetStatus;
+  }
+  throw std::runtime_error("Cannot convert " + pathGetPropertiesAction + " to PathGetPropertiesAction");
+};
+
+class DatalakeRestClient
+{
+public:
+  struct ServiceListFileSystemsOptions
+  {
+    std::string Prefix = std::string();  // Filters results to filesystems within the specified prefix.
+    std::string Continuation = std::string();  // Optional.  When deleting a directory, the number of paths that are deleted with each invocation is limited.  If the number of paths to be deleted exceeds this limit, a continuation token is returned in this response header.  When a continuation token is returned in the response, it must be specified in a subsequent invocation of the delete operation to continue deleting the directory.
+    uint32_t MaxResults = uint32_t();  // An optional value that specifies the maximum number of items to return. If omitted or greater than 5,000, the response will include up to 5,000 items.
```

Comment:
this also feels so weird, it has the same effect as `uint32_t MaxResults = 0;`. But looks so obscure.

Reply:
It is to align with other optional parameter's initialization.